BedrockServer.cpp

```diff
@@ -176,16 +179,27 @@ void BedrockServer::sync(const SData& args,
 
     // Initialize the DB.
     int64_t mmapSizeGB = args.isSet("-mmapSizeGB") ? stoll(args["-mmapSizeGB"]) : 0;
-    SQLite db(args["-db"], args.calc("-cacheSize"), true, args.calc("-maxJournalSize"), -1, workerThreads - 1, args["-synchronous"], mmapSizeGB, args.test("-pageLogging"));
 
-    // And the command processor.
+    // We allocate 25% of the total available file handles to our DB pool. Note that on Linux, the default limit is only
+    // 2014 file handles, so this can be as low as 256 if not adjusted elsewhere (it's adjusted in main.cpp for live
+    // servers).
+    struct rlimit limits;
+    if (getrlimit(RLIMIT_NOFILE, &limits)) {
+        SERROR("Failed to get FD limit");
+    }
+    int fdLimit = limits.rlim_cur / 4;
```

Comment:
To confirm, this effectively limits the number of "infinite threads" to 1/4th the per-process FD limit?

Reply:
Yes, though we can change that limit, and this PR currently sets the limit to 1024 * 10 for live servers.

Reply:
> so we actually have a limit of 10 * perProcessFDLimit / 4

Well, the per process limit for this process is 10x the default. We don't have a limit of FDs of 2.5x the number of filehandles we're allowed to open.